sklearn/utils/_array_api.py

```diff
@@ -105,6 +105,22 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
             yield array_namespace, None, None
 
 
+def _get_namespace_device_dtype_ids(param):
+    """Get pytest parametrization IDs for `yield_namespace_device_dtype_combinations`"""
+    # See #31042 for details
+    try:
+        import array_api_strict
+    except ImportError:
+        return None
```

Comment:
Maybe mention the pytest doc link and mention that returning None doesn't change the default id if array-api-strict is not installed
```suggestion
        return
```

Maybe also quickly mention the fact that this gives better ids for array-api-strict devices?

Reply:
I mentioned that `None` means default but didn't add the link as it was long(er than char limit) and wasn't sure if it is needed, but happy to add if you wish.